server/game/cards/characters/04/balongreyjoy.js

```diff
@@ -0,0 +1,35 @@
+const DrawCard = require('../../../drawcard.js');
+
+class BalonGreyjoy extends DrawCard {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            condition: () => {
+                var otherPlayer = this.game.getOtherPlayer(this.controller);
+
+                return !otherPlayer ||
+                    (this.game.currentChallenge &&
+                    this.game.currentChallenge.challengeType !== 'military' &&
```

Comment:
Balon should only get his effect on military challenges, rather than on all but military challenges.